include/event2/http.h

```diff
@@ -674,7 +674,9 @@ enum evhttp_request_kind { EVHTTP_REQUEST, EVHTTP_RESPONSE };
  *     specified host name resolution will block.
  * @param bev a bufferevent to use for connecting to the server; if NULL, a
  *     socket-based bufferevent will be created.  This bufferevent will be freed
- *     when the connection closes.  It must have no fd set on it.
+ *     when the connection closes.  If a fd is already set on the bufferevent,
```

Comment:
Such API looks too tricky, I guess it will be better a separate method for this

Reply:
Added a function `evhttp_connection_base_bufferevent_set_new` into the interface